### PR TITLE
Channels Bump for 0.15

### PR DESCRIPTION
**Pull request by chrismccord**
@lancehalvorsen ready for review/edits

### CODE REVIEW

File: C_routing.md [hunk 1]
```diff
@@ -778,113 +778,50 @@ end
...
-$ mix phoenix.routes
-. . .
-web_socket_path  GET   /ws       Phoenix.Transports.WebSocket.upgrade/2
-web_socket_path  POST  /ws       Phoenix.Transports.WebSocket.upgrade/2
-long_poller_path  GET   /ws/poll  Phoenix.Transports.LongPoller.poll/2
-long_poller_path  POST  /ws/poll  Phoenix.Transports.LongPoller.publish/2
-```
-
-Notice that our socket definition expands out to four paths with two separate transport mechanisms - WebSockets and LongPolling. If we wanted to make sure that our channel is handled by only one type of transport, we could specify that using the `via` option, like this.
+Phoenix abstracts the socket transport layer and includes two transport out of the box - WebSockets and Long-Polling. If we wanted to make sure that our channel is handled by only one type of transport, we could specify that using the `via` option, like this.
```
Comment (lancehalvorsen) [inline C_routing.md]: Maybe change "two transport" to "two transport mechanisms" or just "two transports"?

File: C_routing.md [hunk 2]
```diff
@@ -778,113 +778,50 @@ end
 
 Channels are a very exciting, real-time component of the Phoenix framework. Channels handle incoming and outgoing messages broadcast over a socket for a given topic. Channel routes, then, need to match requests by socket and topic in order to dispatch to the correct channel. (For a more detailed description of channels and their behavior, please see the [Channel Guide](http://www.phoenixframework.org/docs/channels).)
 
-We define sockets in the router with a block, a path to the socket's mount point, and (optionally) the name of our application to fully qualify our channel module name. This should look familiar after seeing our scope definitions above.
-
-Here's what that looks like in our router file:
+We mount socket handlers in our endpoint, who handle authentication callbacks and channel routes.
```
Comment (lancehalvorsen) [inline C_routing.md]: How about this?

We mount socket handlers in our endpoint. Socket handlers take care of authentication callbacks and channel routes.

File: G_channels.md [hunk 3]
```diff
@@ -5,20 +5,22 @@ Since Elixir is based on message passing, you may wonder why we need this extra
...
 #### The Moving Parts
+
+- Socket Handlers
+
+Phoenix holds a single connection to the server and mutiplexes your channel sockets over that one connection. Socket handlers, such as your `web/channels/user_socket.ex`, are modules that authenticate and identify a socket connection and allow you to set default socket assigns for use in all channels.
+
 - Channel Routes
 
-These are defined in the `router.ex` file as all other routes are. They match on the topic string and dispatch matching requests to the given Channel module. The star character `*` acts as a wildcard matcher, so in the following example route, requests for `sample_topic:pizza` and `sample_topic:oranges` would both be dispatched to the `SampleTopicChannel`.
+These are defined in Socket handlers, such as your `web/channels/user_socket.ex` file as all other routes are. They match on the topic string and dispatch matching requests to the given Channel module. The star character `*` acts as a wildcard matcher, so in the following example route, requests for `sample_topic:pizza` and `sample_topic:oranges` would both be dispatched to the `SampleTopicChannel`.
```
Comment (lancehalvorsen) [inline G_channels.md]: I think we need to get rid of the "as all other routes are." part here. Perhaps we can add something like this:

. . . file, which makes them distinct from other routes.

**Comment by lancehalvorsen** [comment]
@chrismccord, I made a few small changes, rebased, and pushed. Here we go!

**Comment by chrismccord** [comment]
<3 <3 <3

**Pull request merged**